### PR TITLE
Atom pairs wrapper

### PR DESCRIPTION
**Pull request by Boranadas**

### CODE REVIEW

**Comment by Boranadas** [comment]
@greglandrum @NadineSchneider @AndreaVolkamer
This PR includes information string for fingerprint generator, additional fingerprint output types implementation and python wrappers for fingerprint generator and invariant generators. I will make changes on this one based on comments and move on to the next PR for Morgan fingerprint.

Example information string using ":" as seperator:

Common arguments	countSimulation=1 foldedSize=2048 : AtomPairArguments includeChirality=0 use2D=1 minDistance=1 maxDistance=30 : AtomPairEnvironmentGenerator : AtomPairInvariantGenerator includeChirality=0 : No bond invariants generator

File: Code/GraphMol/Fingerprints/AtomPairGenerator.cpp [hunk 1]
```diff
@@ -85,34 +85,79 @@ std::uint32_t getAtomPairCode(std::uint32_t codeI, std::uint32_t codeJ,
...
+
 std::uint32_t AtomPairAtomEnv::getBitId(
     FingerprintArguments *arguments,
     const std::vector<std::uint32_t> *atomInvariants,
     const std::vector<std::uint32_t> *bondInvariants,
     const AdditionalOutput *additionalOutput) const {
+  PRECONDITION((atomInvariants->size() >= d_atomIdFirst) &&
+                   (atomInvariants->size() >= d_atomIdSecond),
+               "bad atom invriants size");
+
```
Comment (NadineSchneider) [inline Code/GraphMol/Fingerprints/AtomPairGenerator.cpp:144]: Small typo in the precondition message

File: Code/GraphMol/Fingerprints/Wrap/FingerprintGeneratorWrapper.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                                                 const int confId) const;
+
+  FingerprintGeneratorWrapper();
+  ~FingerprintGeneratorWrapper();
+};
+
+}  // namespace FingerprintWrapper
+}  // namespace RDKit
+
+#endif
```
Comment (NadineSchneider) [inline Code/GraphMol/Fingerprints/Wrap/FingerprintGeneratorWrapper.h:60]: Again something minor, I wouldn't end the name of all functions wrapped to python with 'Wrapper'

**Review by NadineSchneider — COMMENTED** [review]
In my opinion these changes look good! I didn't find any important things which need to be changed.

File: Code/GraphMol/Fingerprints/Wrap/FingerprintGeneratorWrapper.cpp [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+      "AtomInvGeneratorWrapper", python::init<AtomInvGeneratorWrapper>());
+
+  python::class_<BondInvGeneratorWrapper>(
+      "BondInvGeneratorWrapper", python::init<BondInvGeneratorWrapper>());
+
+  docString = "";
+  python::class_<FingerprintGeneratorWrapper>(
+      "FingerprintGeneratorWrapper",
+      python::init<FingerprintGeneratorWrapper>())
+      .def("getFingerprint", &FingerprintGeneratorWrapper::getFingerprint,
```
Comment (greglandrum) [inline Code/GraphMol/Fingerprints/Wrap/FingerprintGeneratorWrapper.cpp:158]: we start Python method names with capital letters. It's inconsistent with C++, but that's the way we've always done it.

**Pull request closed without merging**